userspace/libsinsp/socket_handler.h

```diff
@@ -390,6 +390,10 @@ class socket_data_handler
 			rec = recv(m_socket, &buf[0], count, 0);
 			switch (rec) {
 			case 0:
+				// When the connection is closed we reset the connection state of the handler.
+				// It could happen when the api-server throttles our requests.
+				m_connecting = false;
+				m_connected = false;
```

Comment:
Maybe a stupid question: Do we have to do the same stuff also in `get_all_data_secure` or it is enough to do it into `get_all_data_unsecure`?

Reply:
That's a leftover from previous experiments, nice catch!